code/obj/critter/misc.dm

```diff
@@ -487,23 +489,54 @@
 			else
 				continue
 
+	attackby(obj/item/W as obj, mob/M as mob)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && !(M in src.friends))
+			if(prob(20))
+				src.visible_message("<span class='notice'>[src] chitters happily at the [W], it will no longer attack [M]!</span>")
+				friends += M
+				playsound(src.loc, "sound/misc/bugchitter.ogg", 50, 0)
+				M.drop_item()
+				qdel(W)
+				src.task = "thinking"
+				src.health = min(src.max_health, src.health + health_gain_from_food)
+				eat_twitch(src)
```

Comment:
qdel and these two lines are common among both branches of the if, you can put them one indentation level lower after the if and else instead of duplicating them.